dolfyn/adp/turbulence.py

```diff
@@ -0,0 +1,753 @@
+                    'time': time},
+            attrs={'units': 'm2 s-2',
+                   'long_name': 'Reynolds Stress Vector',
+                   'standard_name': 'specific_reynolds_stress_of_sea_water'})
+
+        return stress_vec
+
+    def calc_stress_5beam(self, ds, noise=None, orientation=None, beam_angle=25, tke_only=False):
+        """Calculate the stresses from the covariance of along-beam 
+        velocity measurements
+
+        Parameters
+        ----------
+        ds : xarray.Dataset
+          Raw dataset in beam coordinates
+        noise : int or xarray.DataArray, default=0 (time)
+          Doppler noise level in units of m/s
+        orientation : str, default=ds.orientation
+          Direction ADCP is facing ('up' or 'down')
+        beam_angle : int, default=25
+          ADCP beam angle in units of degrees
+        tke_only : bool, default=False
+
+        Returns
+        -------
+        tke_vec(, stress_vec) : xarray.DataArray or tuple[xarray.DataArray]
+          If tke_only is set to False, function returns `tke_vec` and `stress_vec`.
+          Otherwise only `tke_vec` is returned
+
+        Notes
+        -----
+        Assumes small-angle approximation is applicable.
+
+        Assumes ADCP instrument coordinate system is aligned with principal flow
+        directions, i.e. u', v' and w' are aligned to the instrument's (XYZ) 
+        frame of reference.
+
+        The stress equations here utilize u'v'_ to account for small variations
+        in pitch and roll. u'v'_ cannot be directly calculated by a 5-beam ADCP,
+        so it is approximated by the covariance of `u` and `v`. The uncertainty
+        introduced by using this approximation is small if deviations from pitch
+        and roll are small (<5-10 degrees).
+
+        Dewey, R., and S. Stringer. "Reynolds stresses and turbulent kinetic
+        energy estimates from various ADCP beam configurations: Theory." J. of
+        Phys. Ocean (2007): 1-35.
+
+        Guerra, Maricarmen, and Jim Thomson. "Turbulence measurements from 
+        five-beam acoustic Doppler current profilers." Journal of Atmospheric 
+        and Oceanic Technology 34.6 (2017): 1267-1284.
+        """
+
+        warnings.warn("    The 5-beam TKE/stress equations assume the instrument's "
+                      "(XYZ) coordinate system is aligned with the principal "
+                      "flow directions.")
+        if noise is None:
+            noise = 0
+            warnings.warn(
+                '    No "noise" input supplied. Consider calculating "noise" using `calc_doppler_noise`')
+
+        if 'vel_b5' not in ds.data_vars:
+            raise Exception("Must have 5th beam data")
+        if 'beam' not in ds.coord_sys:
+            warnings.warn("    Raw dataset must be in the 'beam' coordinate system. "
+                          "Rotating raw dataset...")
+            ds.velds.rotate2('beam')
+
+        b_angle = getattr(ds, 'beam_angle', beam_angle)
```

Comment:
I think this is way too important to just use 25 deg here if 'beam_angle' isn't found in ds. I don't think we should be assuming a value. I think we should just be throwing an error when it's not in the ds.

Reply:
At the very least there needs to be a "warning using 25 deg beam angle".

Reply:
Erroring out works for me

Reply:
Def error out if you don't find it someplace automatically. Also, including `beam_angle=None` in these calc_stress_ functions seems reasonable (i.e., user can specify this to avoid error), makes sense